File: C++/Matrices2Dy3D/Orden_colum2D.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {

    int xs = 3, ys = 3;
    //vector con (tamaño)
    vector<int> arr(xs * ys, 0);

    //Declaracion de array2d (3*3)
    int Array2d[3][3] = {
        {1, 2, 3},
        {4, 5, 6},
        {7, 8, 9}
    };

    //formula de acomodo de Array2d en vector (arr)
    int k = 0;
        for (int x = 0; x < xs; x++) 
        {
            for (int y = 0; y < ys; y++) 
            {
                k = x * xs + y;
                arr[k] = Array2d[x][y];
                k = k + 1;
            }
        }

    cout << "Los elementos del array bidimencional son: ";
    cout << endl;
    
    for (auto& row : Array2d) {          //Recorre renglones
        for (int element : row) {        //Recorre los elementos de cada renglon
            cout << element << " ";
        }
        cout << endl;
    }

    //Recorrido del vector final
    cout << "Los elementos del array unidimensional son: ";
    for (int x = 0; x < xs; x++) {
        for (int y = 0; y < ys; y++) {
            cout << arr[x * xs + y] << " ";
        }
    }
    cout << endl;
}

File: JavaScript/Matrices2Dy3D/Orden_colum2D.js
//Declaracion del arreglo donde voy a vaciar el array2d
const xs = 3, ys = 3;
//funcion fill() para rellenar con 0 por mientras
const arr = new Array(xs * ys).fill(0);

const Array2d = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];

//formula para vaciar array 2d en fila 1d
let k = 0;
for (let x = 0; x < xs; x++) {
    for (let y = 0; y < ys; y++) {
        k = x * xs + y;
        arr[k] = Array2d[x][y];
        k = k + 1;
    }
}

//Arreglo cuadradito
console.log("Los elementos del array bidimensional son: ");
for (const row of Array2d) {
    console.log();
    for (const ele of row) {
        console.log(ele + " ");
    }
    console.log();
}

//arreglo en fila 1d
console.log("Los elementos del array unidimensional son: ");
for (let x = 0; x < xs; x++) {
    for (let y = 0; y < ys; y++) {
        console.log(arr[x * xs + y] + " ");
    }
}